### PR TITLE
Add Layer0

### PR DESCRIPTION
**Pull request by tristanlee85**

### CODE REVIEW

File: layer0/api/db/schema.prisma [hunk 1]
```diff
@@ -5,7 +5,7 @@ datasource db {
 
 generator client {
   provider      = "prisma-client-js"
-  binaryTargets = "native"
+  binaryTargets = ["native","rhel-openssl-1.0.x"]
```
Comment (thedavidprice) [inline layer0/api/db/schema.prisma:8]: I'm going to try with just native unless you know definitively that won't work.

Comment (thedavidprice) [inline layer0/api/db/schema.prisma:8]: I'm going to try with just native unless you know definitively that won't work.

File: layer0/routes.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// This file was automatically added by layer0 init.
+// You should commit this file to source control.
+import { Router } from '@layer0/core'
+import { redwoodRoutes } from '@layer0/redwood'
+
+export default new Router().use(redwoodRoutes)
```
Comment (thedavidprice) [inline layer0/routes.js:6]: Curious about what this does. Are there Docs or a quick explanation?

Comment (thedavidprice) [inline layer0/routes.js:6]: Curious about what this does. Are there Docs or a quick explanation?

Comment (thedavidprice) [reply layer0/routes.js:6]: Ah, found this: https://docs.layer0.co/guides/redwoodjs

Comment (thedavidprice) [reply layer0/routes.js:6]: Ah, found this: https://docs.layer0.co/guides/redwoodjs

**Review by thedavidprice — CHANGES_REQUESTED** [review]
@tristanlee85 I sent you a direct message with video for further discussion. I'm understanding much better what's supposed to be going on, but I'm running into performance issues (navigating pages is crashing site) and Auth is broken.

Look for my message in Slack with full details.

If we need, we can merge this PR to see how the CI is working and continue from there. I did make a change to schema.prisma --> none of our other deployments for AWS Lambdas require anything but `binaryTargets = "native"`, although that wasn't always the case. Just let me know if I'm missing some information.

File: .github/workflows/build-test.yaml [hunk 3]
```diff
@@ -286,4 +286,60 @@ jobs:
         working-directory: ${{ steps.project-directory.outputs.dir }}
         run: yarn rw test
         env:
-          TEST_DATABASE_URL: postgresql://test:password@localhost:5432/test                    
\ No newline at end of file
+          TEST_DATABASE_URL: postgresql://test:password@localhost:5432/test  
+
+  layer0:
```
Comment (thedavidprice) [inline .github/workflows/build-test.yaml:291]: CI for the project; required for deployment action to run

File: .github/workflows/deploy-layer0.yml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    if: github.repository == 'redwoodjs/deploy-target-ci'
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16'
```
Comment (thedavidprice) [inline .github/workflows/deploy-layer0.yml:18]: This shouldn't have anything to do with built functions (unless I misunderstand how layer0 build is working). Redwood will build the API functions for Node.js v12, which we'll bump to v14 in a few months.

Just let me know if there's a reason this won't work.

File: layer0/api/db/schema.prisma [hunk 5]
```diff
@@ -5,7 +5,7 @@ datasource db {
 
 generator client {
   provider      = "prisma-client-js"
-  binaryTargets = "native"
+  binaryTargets = ["native"]
```
Comment (thedavidprice) [inline layer0/api/db/schema.prisma:8]: I'm going to try with just native unless you know definitively that won't work.

File: .github/workflows/deploy-layer0.yml (hunk 4 above)
Comment (tristanlee85) [reply .github/workflows/deploy-layer0.yml:18]: When Layer0 does its build of the router handler function, it's possible there could be issues with having that handler (not the RW functions) built with Node 16, but then run under Node 14 in AWS. We can try it with Node 16 though and see if we identify any issues.

File: layer0/api/db/schema.prisma (hunk 5 above)
Comment (tristanlee85) [reply layer0/api/db/schema.prisma:8]: I recall getting this error again during this project too. I'm not sure if it is related to GH action doing a native build that then differs from the env in AWS. The build will be successful, but requests that use the API may fail. Again, we can try this just as native to get the workflow to run and adjust later if needed.

**Review by thedavidprice — DISMISSED** [review]

**Pull request merged**